server/base/src/test/java/org/apache/accumulo/server/conf/CheckCompactionConfigTest.java

```diff
@@ -128,6 +128,57 @@ public void testNoPlanner() throws Exception {
     assertTrue(e.getMessage().startsWith(expectedErrorMsg));
   }
 
+  @Test
+  public void testRepeatedCompactionExecutorID() throws Exception {
+    String inputString = ("tserver.compaction.major.service.cs1.planner="
+        + "org.apache.accumulo.core.spi.compaction.DefaultCompactionPlanner \n"
+        + "tserver.compaction.major.service.cs1.planner.opts.executors=\\\n"
+        + "[{'name':'small','type':'internal','maxSize':'16M','numThreads':8},\\\n"
+        + "{'name':'medium','type':'internal','maxSize':'128M','numThreads':4},\\\n"
+        + "{'name':'small','type':'internal','numThreads':2}]").replaceAll("'", "\"");
+    String expectedErrorMsg = "Duplicate Compaction Executor ID found";
+
+    String filePath = writeToFileAndReturnPath(inputString);
+
+    var e = assertThrows(IllegalStateException.class,
+        () -> CheckCompactionConfig.main(new String[] {filePath}));
+    assertTrue(e.getMessage().startsWith(expectedErrorMsg));
+  }
+
+  @Test
+  public void testInvalidTypeValue() throws Exception {
+    String inputString = ("tserver.compaction.major.service.cs1.planner="
+        + "org.apache.accumulo.core.spi.compaction.DefaultCompactionPlanner \n"
+        + "tserver.compaction.major.service.cs1.planner.opts.executors=\\\n"
+        + "[{'name':'small','type':'internal','maxSize':'16M','numThreads':8},\\\n"
+        + "{'name':'medium','type':'internal','maxSize':'128M','numThreads':4},\\\n"
+        + "{'name':'large','type':'internl','numThreads':2}]").replaceAll("'", "\"");
+    String expectedErrorMsg = "type must be 'internal' or 'external'";
+
+    String filePath = writeToFileAndReturnPath(inputString);
+
+    var e = assertThrows(IllegalArgumentException.class,
+        () -> CheckCompactionConfig.main(new String[] {filePath}));
+    assertTrue(e.getMessage().startsWith(expectedErrorMsg));
```

Comment:
In this particular test class, testing of the message text is used for each of the error checks.The two new tests follow the existing format for the class. I was working with the new compaction service and had an error in my config. I ran it through the check, but the message gave no real information as to what was wrong. It wasn't until I manually found it that I was able to correct it. The additional messaging will hopefully make error discovery easier in the future.

Reply:
Sorry if I am being unclear.  I like the additions to the error message.  Testing the exception is thrown is good.

My question directly concerns testing asserting the message starts with a specific string in the tests. The remainder of the test conditions are are good addition - just the two message starts with bit stuck me as something I don't recall seeing normally in other tests.

Reply:
No problem. I understood. I was just trying to point out that if you look at that test class in particular, all of the other error condition checks also check the message text. So I decided to follow the same format for the two additional tests. 
Also, since the purpose of the additions was to specifically add output new information to the user, I'm wasn't sure of a better way to verify the new messages appeared in the error output.

Reply:
Sorry - didn't realize that the test was doing that already - all good.